torch/_dynamo/external_utils.py

```diff
@@ -187,3 +187,35 @@ def call_module_hooks_from_backward_state(
         if new_result is not None:
             result = new_result
     return result
+
+
+# used by dont_skip_tracing decorator to ignore trace rules recursively
+_ignore_skip_function_variable = False
```

Comment:
Can this be done without using global state?

Reply:
How can we implement this without a global in the case of a skipped function? e.g.
```python
@dont_skip_tracing
def f():
    # do something that forces a skipped frame
    g()

def g():
    ...
```

We need to somehow remember during our call to `g` that `dont_skip_tracing` is still active.

My idea for doing this without a global would be to take advantage of wrapper detection (i.e. something similar to what I'm doing on the non-recursive disable PR). We can propagate `dont_skip_tracing` when handling normal graph breaks by wrapping the resume function, but I don't think this approach will work on the above case.

Reply:
Do you think an approach of looking for a `dont_skip_tracing` wrapper function in the traceback would be better?